### PR TITLE
fix: Revert all breaking changes from 1.0.0, deprecate sourceMap

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: packages/worker/src/workflow/bundler.ts [hunk 1]
```diff
@@ -26,7 +26,12 @@ export function moduleMatches(userModule: string, modules: string[]): boolean {
   return modules.some((module) => userModule === module || userModule.startsWith(`${module}/`));
 }
 
-export interface WorkflowBundle {
+export interface WorkflowBundleWithSourceMap {
+  /**
+   * Source maps are genreated inline, this is no longer used
```
Comment (lorensr) [inline packages/worker/src/workflow/bundler.ts:31]:
```suggestion
   * Source maps are generated inline—this is no longer used
```

File: packages/worker/src/workflow/bundler.ts [hunk 2]
```diff
@@ -102,6 +107,7 @@ export class WorkflowCodeBundler {
 
     // Cast because the type definitions are inaccurate
     return {
+      sourceMap: '',
```
Comment (lorensr) [inline packages/worker/src/workflow/bundler.ts:110]:
```suggestion
      sourceMap: 'this field is deprecated',
```

I'm envisioning devs logging this value, seeing it empty, and wondering what they did wrong.

**Review by lorensr — CHANGES_REQUESTED** [review]
this PR:
```
export interface WorkflowBundlePath {
  codePath: string;
}
export type WorkflowBundleOption = WorkflowBundle | WorkflowBundlePath;
```

`1.0.1`:
```
export interface WorkflowBundlePathWithSourceMap {
    codePath: string;
    sourceMapPath: string;
}
export declare type WorkflowBundleOption = WorkflowBundleWithSourceMap | WorkflowBundlePathWithSourceMap;
```

I think it's okay for `WorkflowBundleWithSourceMap` to change name to `WorkflowBundle` because it has the same fields, but think we need to add `sourceMapPath: string;`?

File: packages/worker/src/worker.ts [hunk 3]
```diff
@@ -426,7 +422,7 @@ export class Worker {
     const connection = options.connection ?? (await InternalNativeConnection.connect());
     let nativeWorker: NativeWorkerLike;
     try {
-      nativeWorker = await nativeWorkerCtor.create(connection, compiledOptions);
+      nativeWorker = await nativeWorkerCtor.create(connection, compiledOptions, bundle?.code);
```
Comment (mjameswh) [inline packages/worker/src/worker.ts:425]: I had miss that previously, but we should not need to pass code to native worker. Not blocking though.

Comment (bergundy) [reply packages/worker/src/worker.ts:425]: Yeah, I missed this too TBH, it's for generating the buildId, I think we should do this in the worker factory.

File: packages/worker/src/workflow/bundler.ts (hunk 2 above)
Comment (bergundy) [reply packages/worker/src/workflow/bundler.ts:110]: Good suggestion.

**Comment by bergundy** [comment]
I'll rename the second interface, will drop the internal type alias to verify the docs aren't referencing it accidentally.

File: packages/worker/src/worker.ts (hunk 3 above)
Comment (bergundy) [reply packages/worker/src/worker.ts]: Refactored this a bit.

File: packages/worker/src/workflow/bundler.ts [hunk 4]
```diff
@@ -26,7 +26,12 @@ export function moduleMatches(userModule: string, modules: string[]): boolean {
   return modules.some((module) => userModule === module || userModule.startsWith(`${module}/`));
 }
 
-export interface WorkflowBundle {
+export interface WorkflowBundleWithSourceMap {
+  /**
+   * Source maps are genreated inline - this is no longer used
```
Comment (lorensr) [inline packages/worker/src/workflow/bundler.ts:31]:
```suggestion
   * Source maps are generated inline - this is no longer used
```

File: packages/worker/src/worker-options.ts [hunk 5]
```diff
@@ -18,7 +18,27 @@ export type { WebpackConfiguration };
...
+
+/**
+ * Note this no longer contains a source map.
+ * The name was preserved to avoid breaking backwards compatibility.
+ *
+ * @deprecated
+ */
+export interface WorkflowBundlePathWithSourceMap {
+  codePath: string;
+  sourceMap: string;
```
Comment (lorensr) [inline packages/worker/src/worker-options.ts:30]:
```suggestion
  sourceMapPath: string;
```

**Review by lorensr — APPROVED** [review]

File: packages/worker/src/worker-options.ts (hunk 5 above)
Comment (bergundy) [reply packages/worker/src/worker-options.ts]: Thanks!

**Pull request merged**